serviceconnector/src/createServiceConnector/TargetServiceTypeStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { INewStorageAccountDefaults, StorageAccountKind, StorageAccountListStep, StorageAccountPerformance, StorageAccountReplication } from "@microsoft/vscode-azext-azureutils";
+import { AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { TargetServiceType, TargetServiceTypeName } from "../../constants";
+import { ICreateLinkerContext } from "./ICreateLinkerContext";
+
+export class TargetServiceTypeStep extends AzureWizardPromptStep<ICreateLinkerContext>{
```

Comment:
I think that this should be `StorageServiceTypeStep` or something similar. I think ideally, `TargetServiceTypeStep` would be broader categories, like `Azure Storage`, `Databases`, `Key Vault`.

Then we would have the specific service type steps for each target service. I know right now that we are only supporting storage, but we should design it with expanding in mind.

Reply:
I think I would still push to have `target` in the name still. I think we could remove type and make it `TargetServiceStep`, or `TargetStorageServiceStep`. Having target in there just adds a lot of much-needed context for what's the user is picking in the step.

Reply:
`StorageTargetServiceTypeStep`?

Reply:
I had originally thought TargetServiceTypeStep would branch out in the subwizard to database types (Cosmos DB, PostGreSql, etc.). 

Currently the portal lists out each service type like so: 
![image](https://github.com/microsoft/vscode-azuretools/assets/59709511/a018c3b6-a754-42a5-8c4f-50e5664003b2)
So I was following that schema but based on what you proposed Nathan we would split this into multiple steps?

Reply:
Based on offline conversation will be keeping this as one step and adding groupings.